skimage/color/colorconv.py

```diff
@@ -135,26 +151,40 @@ def _prepare_colorarray(arr, force_copy=False):
     return _func(arr, force_copy=force_copy)
 
 
-def rgba2rgb(rgba, background=(1, 1, 1)):
+def _validate_channel_axis(channel_axis, ndim):
+    if not isinstance(channel_axis, int):
+        raise np.AxisError("channel_axis must be an integer")
```

Comment:
@grlee77 it would be more consistent if this one were a `TypeError`, from what I tested out for #5420 (see line https://github.com/scikit-image/scikit-image/pull/5420/files#diff-a62ddc4b2d437b0b4ee3a01b2eba9a2cb7a148f9cf03f1e6902e6307a9596e2fR61).

Reply:
Agreed. I have switched this one to `TypeError`